File: UI/fitbook_desktop/lib/models/requests/trainer_update_request.dart
import 'package:json_annotation/json_annotation.dart';

import '../common/api_request_body.dart';

part 'trainer_update_request.g.dart';

@JsonSerializable()
class TrainerUpdateRequest implements ApiRequestBody {
  TrainerUpdateRequest({
    required this.firstName,
    required this.lastName,
    required this.specializationId,
    this.biography,
    this.imageUrl,
    required this.isAvailable,
    required this.isActive,
  });

  final String firstName;
  final String lastName;
  final int specializationId;
  final String? biography;
  final String? imageUrl;
  final bool isAvailable;
  final bool isActive;

  factory TrainerUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$TrainerUpdateRequestFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$TrainerUpdateRequestToJson(this);
}

File: FitBook.Model/Requests/Trainers/TrainerUpdateRequest.cs
namespace FitBook.Model.Requests.Trainers;

public class TrainerUpdateRequest
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public int SpecializationId { get; set; }
    public string? Biography { get; set; }
    public string? ImageUrl { get; set; }
    public bool IsAvailable { get; set; }
    public bool IsActive { get; set; }
}
